File: app/src/main/java/com/drawable/Line.java
package com.drawable;

import android.graphics.Canvas;
import android.graphics.Paint;

import com.util.Color;

import java.math.BigDecimal;

@SuppressWarnings({"unused", "WeakerAccess"})
public class Line extends ComplexDrawable {
    private double lineSize;

    public Line(double x1, double y1, double x2, double y2, double lineSize, int color) {
        this(x1, y1, x2, y2, lineSize, color, 0, 0);
    }

    public Line(double x1, double y1, double x2, double y2, double lineSize, Color color) {
        this(x1, y1, x2, y2, lineSize, color.getInt(), 0, 0);
    }

    public Line(double x1, double y1, double x2, double y2, double lineSize, int color, double xVelocity, double yVelocity) {
        super(x1, y1, x2, y2, color, xVelocity, yVelocity);
        this.lineSize = lineSize;
    }

    public Line(double x1, double y1, double x2, double y2, double lineSize, Color color, double xVelocity, double yVelocity) {
        this(x1, y1, x2, y2, lineSize, color.getInt(), xVelocity, yVelocity);
    }

    public Line(Line line) {
        super(line);
        this.lineSize = line.getLineSize();
    }

    public double getLineSize() {
        return lineSize;
    }

    public void setLineSize(double lineSize) {
        this.lineSize = lineSize;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Line)) return false;
        Line other = (Line) o;
        BigDecimal first = new BigDecimal(Double.toString(this.lineSize));
        BigDecimal second = new BigDecimal(Double.toString(other.lineSize));
        return first.compareTo(second) == 0 && this.getColor() == other.getColor();
    }

    @Override
    public void draw(Canvas canvas, Paint paint) {
        paint.setStyle(Paint.Style.FILL);
        paint.setColor(getColor());
        paint.setStrokeWidth((float) lineSize);
        canvas.drawLine((float) getX(), (float) getY(), (float) getX2(), (float) getY2(), paint);
        animate();
    }
}

File: app/src/main/java/com/drawable/Line.kt
package com.drawable

import android.graphics.Canvas
import android.graphics.Paint

import com.util.Color

import java.math.BigDecimal

@Suppress("unused", "NAME_SHADOWING", "EqualsOrHashCode")
class Line : ComplexDrawable {
    var lineSize: Double = 0.toDouble()

    constructor(x1: Double, y1: Double, x2: Double, y2: Double, lineSize: Double, color: Color) :
            this(x1, y1, x2, y2, lineSize, color.int, 0.0, 0.0)

    constructor(x1: Double, y1: Double, x2: Double, y2: Double, lineSize: Double, color: Int, xVelocity: Double = 0.0,
                              yVelocity: Double = 0.0) : super(x1, y1, x2, y2, color, xVelocity, yVelocity) {
        this.lineSize = lineSize
    }

    constructor(x1: Double, y1: Double, x2: Double, y2: Double, lineSize: Double, color: Color, xVelocity: Double, yVelocity: Double) :
            this(x1, y1, x2, y2, lineSize, color.int, xVelocity, yVelocity)

    constructor(line: Line) : super(line) {
        this.lineSize = line.lineSize
    }

    override fun equals(other: Any?): Boolean {
        if (other !is Line) return false
        val other = other as Line?
        val first = BigDecimal(java.lang.Double.toString(this.lineSize))
        val second = BigDecimal(java.lang.Double.toString(other!!.lineSize))
        return first.compareTo(second) == 0 && this.color == other.color
    }

    override fun draw(canvas: Canvas, paint: Paint) {
        paint.style = Paint.Style.FILL
        paint.color = color
        paint.strokeWidth = lineSize.toFloat()
        canvas.drawLine(x.toFloat(), y.toFloat(), x2.toFloat(), y2.toFloat(), paint)
        animate()
    }

    override fun toString(): String {
        return super.toString() + "\n \tLine size: " + lineSize + "\n"
    }
}